src/provider.ts

```diff
@@ -12,7 +12,9 @@ export const setChainId = (newChainId: ChainID): void => {
 };
 export const getChainId = (): ChainID => chainId;
 
-const provider = new providers.JsonRpcProvider(process.env.CHAIN_RPC_ENDPOINT);
+const provider = new providers.StaticJsonRpcProvider(
+  process.env.CHAIN_RPC_ENDPOINT,
+);
```

Comment:
I'd want @area 's input here, because the way I think about this, we still need the "normal", "noisy" `JsonRpcProvider` in the block ingestor as opposed to the `StaticJsonRpcProvider` one

Reply:
Chatted briefly with Raul on a call, his worry was around listeners firing, but they should still fire with the StaticJsonRpcProvider.